### PR TITLE
Use typed array in TextDecoder::Decode

### LINKED ISSUE #14674: Rewrite TextDecoder::Decode to use typed array API
Instead of using `array_buffer_view_data`, this method should use [ArrayBufferView::from](https://github.com/servo/rust-mozjs/blob/master/src/typedarray.rs#L66) and return an error value if a typed array cannot be extracted from the JS object. We'll need to declare the value using the `typedarray!` macro (see [the tests](https://github.com/servo/rust-mozjs/blob/master/tests/typedarray.rs) for examples).

Code: `components/script/dom/textdecoder.rs`
Tests: `./mach test-wpt tests/wpt/web-platform-tests/encoding/`

Note: this relies on https://github.com/servo/rust-mozjs/pull/323 before the ArrayBufferView type exists.

### PR DESCRIPTION
**Pull request by zaynetro**
Use typed array API in TextDecoder::Decode

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #14674

- [x] These changes do not require tests because no new methods were introduced

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/14727)

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -1246,7 +1246,7 @@ dependencies = [
 [[package]]
 name = "js"
 version = "0.1.3"
-source = "git+https://github.com/servo/rust-mozjs#342f304a455080acf64f5ceb40a7f8059481ca01"
+source = "git+https://github.com/servo/rust-mozjs#15ff1e83446e998112dcde731610e8b60cc32abf"
```
Comment (Wafflespeanut) [inline Cargo.lock]: Hm, the tree already seems to be using this revision. Rebase?

Comment (zaynetro) [inline Cargo.lock]: OK, it wasn't there when pull request was submitted. Will do it now

Comment (zaynetro) [inline Cargo.lock]: @Wafflespeanut done

**Comment by jdm** [comment]
@bors-servo: r+
Thanks for making this change!

**Pull request merged**